crates/globwalk/src/lib.rs

```diff
@@ -0,0 +1,129 @@
+use std::fmt::Display;
+
+use glob_match::glob_match;
+use turborepo_paths::AbsolutePathBuf;
+
+pub enum WalkType {
+    Files,
+    Folders,
+    All,
+}
+
+#[derive(Debug)]
+pub enum WalkError {
+    Error(walkdir::Error),
```

Comment:
Can we do `#[from]` with thiserror? Alternatively, is there any issue with returning the `walkdir::Error` type itself?

Reply:
I think for one variant we can probably forgo that. It is an enum in case we need to add additional stuff later. I usually have an enum as the top level error for non-leaf libraries.